lib/libzfs/libzfs_util.c

```diff
@@ -1940,3 +1941,91 @@ zprop_iter(zprop_func func, void *cb, boolean_t show_all, boolean_t ordered,
 {
 	return (zprop_iter_common(func, cb, show_all, ordered, type));
 }
+
+static void
+nomem(void)
+{
+	(void) fprintf(stderr, gettext("internal error: out of memory\n"));
+	exit(1);
+}
+
+/*
+ * Utility function to guarantee malloc() success.
+ */
+
+static void *
+safe_malloc(size_t size)
+{
+	void *data;
+
+	if ((data = calloc(1, size)) == NULL)
+		nomem();
+
+	return (data);
+}
+
+#ifndef ZFS_VERSION_EXTRA
+#define ZFS_VERSION_EXTRA ""
+#endif
+/*
+ * Returns the zfs userland version with ZFS_META_RELEASE and ZFS_META_GITREV
+ * Returned char * must be freed by the caller.
+ */
+char *
+zfs_version_userland(void)
+{
+	char *version = safe_malloc(256);
+
+	(void) snprintf(version, 256, "%s%s",
+	    ZFS_META_GITREV, ZFS_VERSION_EXTRA);
+
+	return version;
+}
+
+/*
+ * Returns the zfs kernel version read from "/sys/module/zfs/version"
+ * Returned char * must be freed by the caller.
+ */
+char *
+zfs_version_kernel(void)
+{
+	char *f_zversion = "/sys/module/zfs/version";
+	char *notfound = "zfs-kmod version not found";
+	char *version = safe_malloc(64);
```

Comment:
Why is this buffer size different than in zfs_version_userland()?

Reply:
I generally try to be frugal with memory allocations and I noticed the string from "/sys/module/zfs/version" wasn't very long (17) on my system.  What would you recommend?